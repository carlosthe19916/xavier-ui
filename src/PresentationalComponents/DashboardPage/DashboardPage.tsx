import React from 'react';
import { withRouter } from 'react-router-dom';
import asyncComponent from '../../Utilities/asyncComponent';
import './DashboardPage.scss';
import {
    Main,
    PageHeader,
    PageHeaderTitle
} from '@redhat-cloud-services/frontend-components';
import { GlobalProps } from '../../models/GlobalProps';

const EmptyDashboard = asyncComponent(() => import('../EmptyDashboard/EmptyDashboard'));

interface Props extends GlobalProps {
}

interface State {
}

class DashboardPage extends React.Component<Props, State> {

    render() {
        return (
            <React.Fragment>
                <PageHeader>
                    <PageHeaderTitle title='Dashboard' />
                </PageHeader>
                <Main>
                    <EmptyDashboard />
                </Main>
            </React.Fragment>
        );
    }
}

export default withRouter(DashboardPage);
