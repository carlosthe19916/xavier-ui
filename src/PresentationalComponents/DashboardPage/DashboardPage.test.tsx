import { shallow } from "enzyme";
import DashboardPage from "./DashboardPage";

jest.mock("react-router-dom"); // <- Use this line

describe("DashboardPage", () => {
  it("expect to render ", () => {
    const wrapper = shallow(<DashboardPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
