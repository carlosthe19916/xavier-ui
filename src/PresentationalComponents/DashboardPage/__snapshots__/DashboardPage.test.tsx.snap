// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardPage expect to render  1`] = `
<Fragment>
  <PageHeader>
    <PageHeaderTitle
      title="Dashboard"
    />
  </PageHeader>
  <Connect(Main)>
    <EmptyDashboard />
  </Connect(Main)>
</Fragment>
`;
